### PR TITLE
improved definition of done, so its easier to run automated tests againts solution

### PR DESCRIPTION
**Pull request by piotrkowalczuk**
* how it should be submited
* ports that it should expose
* slightly different way to handle timeouts (use specific status code)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,8 +26,9 @@ the result as the list of integers.
 The endpoint needs to return the result as quickly as possible, but always 
 within 500 milliseconds. It needs to be able to deal with error conditions when 
 retrieving the URLs. If a URL takes too long to respond, it must be ignored. It 
-is valid to return an empty list as result only if all URLs returned errors or 
-took too long to respond.
+is valid to return an empty list as result only if all URLs returned errors.
+
+If server is not able to handle requests in given time it should return `http.StatusRequestTimeout` status code.
```
Comment (azhukovta) [inline README.md]: I think they should always respond with []numbers array even if it's empty. Well, the origin of this task is DSP handling bid requests. They send us bid request, we process it and respond. DSP always replies with valid answer even if it is empty array.

Comment (piotrkowalczuk) [reply README.md]: @azhukovta Reason for this change is to be able to easier read vegeta graphs. To know when response was to late but code knows about it and when it replies to late but from code perspective response is in time.

Comment (azhukovta) [reply README.md]: I don't think the code should know he is too late. I mean code should Never be too late, right?

File: README.md [hunk 2]
```diff
@@ -76,13 +77,20 @@ Solve the task described above using Go. Only use what's provided in the Go
...
 describes the intentions and rationale behind your solution. Also write down 
 any ambiguities that you see in the task description, and describe you how you 
-interpreted them and why. If applicable, write automated tests for your code.
+interpreted them and why.
 
-For testing purposes, you will be provided with an example server that, when 
+As a reference, you will be provided with an example server that, when 
 run, listens on port 8090 and provides the endpoints /primes, /fibo, /odd and 
-/rand.
+/rand. __Another (more demanding) environment will be used to evaluate your solution__.
```
Comment (azhukovta) [inline README.md]: This is a hint, but it does not help the candidate in any way. If you want to force them use context timeout, just say something like "Make sure your code is able to handle really large arrays of numbers"

Comment (piotrkowalczuk) [reply README.md]: I don't want to help candidate to pass. I want to find good people.

File: README.md (hunk 1 above)
Comment (piotrkowalczuk) [reply README.md]: I think its completely other way around. Everywhere (besides bidder) server should know when it is not able to give response in given time. That entire concept of status codes.

File: README.md (hunk 2 above)
Comment (azhukovta) [reply README.md]: What is the purpose of this sentence?

Comment (piotrkowalczuk) [reply README.md]: Together witch change above `As a reference,` it should be clear that only by checking if his solution works against given server does not mean that he will pass internal tests.

File: README.md (hunk 1 above)
Comment (azhukovta) [reply README.md]: Okay, but if you want some other task, you can create your own task. But this task is a RTB miniature. And it shows how people think toward uptime, fault tolerance. Which is significant for RTB and I think these skills help in other projects in TA as well

File: README.md (hunk 2 above)
Comment (azhukovta) [reply README.md]: Okay, I think it is too much to the review. Lets discuss it at the go stack team meeting with everyone. And come to a solution together

File: README.md (hunk 1 above)
Comment (piotrkowalczuk) [reply README.md]: This that does not check anything of that. It only replicate exact use case from the bidder. Using status codes is not against fault tolerance or uptime. Its about communication.

Comment (trinitum) [reply README.md]: @piotrkowalczuk request timeout is sent when *client* failed to submit request within expected time, if server can't produce request within set time limits it should return 503 or some other 5xx error

Comment (piotrkowalczuk) [reply README.md]: @trinitum good point!

**Comment by piotrkowalczuk** [comment]
Lets split it into multiple smaller PRs that will address the [proposal](https://travelaudience.atlassian.net/wiki/spaces/GST/pages/704446631/Development+environment+for+applicants+and+clarified+task+description).

**Pull request closed without merging**